domains/pallets/messenger/src/lib.rs

```diff
@@ -919,33 +921,32 @@ mod pallet {
                 TransactionValidityError::Invalid(InvalidTransaction::BadProof),
             )?;
 
-            // TODO: system domain has been removed
-            // on system domain, ensure the core domain info is at K-depth and state root matches
-            // if T::SelfDomainId::get().is_system() {
-            // if let Some((domain_id, block_info, state_root)) =
-            // extracted_state_roots.core_domain_info.clone()
-            // {
-            // // ensure the block is at-least k-deep
-            // let confirmed = T::DomainInfo::domain_best_number(domain_id)
-            // .and_then(|best_number| {
-            // best_number
-            // .checked_sub(&T::ConfirmationDepth::get())
-            // .map(|confirmed_number| confirmed_number >= block_info.block_number)
-            // })
-            // .unwrap_or(false);
-            // ensure!(confirmed, InvalidTransaction::BadMandatory);
-
-            // // verify state root of the block
-            // let valid_state_root = T::DomainInfo::domain_state_root(
-            // domain_id,
-            // block_info.block_number,
-            // block_info.block_hash,
-            // )
-            // .map(|got_state_root| got_state_root == state_root)
-            // .unwrap_or(false);
-            // ensure!(valid_state_root, InvalidTransaction::BadMandatory)
-            // }
-            // }
+            // on consensus, ensure the domain info is at K-depth and state root matches
+            if T::SelfChainId::get().is_consensus_chain() {
+                if let Some((domain_id, block_info, state_root)) =
+                    extracted_state_roots.domain_info.clone()
+                {
+                    // ensure the block is at-least k-deep
+                    let confirmed = T::DomainInfo::domain_best_number(domain_id)
+                        .and_then(|best_number| {
+                            best_number
+                                .checked_sub(&T::ConfirmationDepth::get())
+                                .map(|confirmed_number| confirmed_number >= block_info.block_number)
+                        })
+                        .unwrap_or(false);
+                    ensure!(confirmed, InvalidTransaction::BadMandatory);
```

Comment:
Perhaps we should use `HeadDomainNumber` (`HeadDomainNumber` if there is an ongoing domain block) as the `domain_best_number` instead of `HeadReciptNumber` because `HeadReciptNumber` may decrease after processing fraud proof.

Assume `ConfirmationDepth` is 100, an XDM is initialed at domain block 100 when the XDM is relayed by the src_chain relayer `HeadReciptNumber` is 200, and then a fraud proof is submitted and reverts the `HeadReciptNumber` to 190, after that the dst_chain receives the XDM and find `confirmed_number` is 90 now, thus rejecting the XDM.